src/python/essentia/pytools/extractors/melspectrogram.py

```diff
@@ -0,0 +1,227 @@
+# Copyright (C) 2006-2020  Music Technology Group - Universitat Pompeu Fabra
+#
+# This file is part of Essentia
+#
+# Essentia is free software: you can redistribute it and/or modify it under
+# the terms of the GNU Affero General Public License as published by the Free
+# Software Foundation (FSF), either version 3 of the License, or (at your
+# option) any later version.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
+# details.
+#
+# You should have received a copy of the Affero GNU General Public License
+# version 3 along with this program. If not, see http://www.gnu.org/licenses/
+
+import argparse
+import os
+
+import numpy as np
+
+from essentia import Pool
+from essentia import run
+
+from essentia.streaming import (MonoLoader, FrameCutter, Windowing, Spectrum,
+                                MelBands, UnaryOperator)
+
+ZERO_PADDING = 0
+WINDOW_TYPE = 'hann'
+FRAME_SIZE = 1024
+HOP_SIZE = 512
+NUMBER_BANDS = 24
+SAMPLE_RATE = 44100.
+LOW_FREQUENCY_BOUND = 0.
+HIGH_FREQUENCY_BOUND = 22050.
+WARPING_FORMULA = 'htkMel'
+WEIGHTING = 'warping'
+NORMALIZE = 'unit_sum'
+BANDS_TYPE = 'power'
+COMPRESSION_TYPE = 'shift_scale_log'
+
+
+def melspectrogram(filename, npy_file=None, force=False, verbose=False, sample_rate=SAMPLE_RATE, frame_size=FRAME_SIZE,
```

Comment:
Note that it is possible to use `**kwargs`  for non-essential parameters like here: https://github.com/MTG/essentia/blob/master/src/python/essentia/pytools/spectral.py#L250

Reply:
Right, the function definition looks ugly with so many parameters. However, I decided to explicitly mention all the parameters because I wanted `help(melspectrogram)` to show all the parameters available with their default values. Especially because some parameters (e.g., `compression_type`) doesn't match any parameter from Essentia algorithms directly. 
Do you think there is a better way to achieve this?

```
melspectrogram?
```

```ShellSession
Signature:
melspectrogram(
    filename,
    npy_file=None,
    force=False,
    verbose=False,
    sample_rate=44100.0,
    frame_size=1024,
    hop_size=512,
    window_type='hann',
    zero_padding=0,
    low_frequency_bound=0.0,
    high_frequency_bound=22050.0,
    number_bands=24,
    warping_formula='htkMel',
    weighting='warping',
    normalize='unit_sum',
    bands_type='power',
    compression_type='shift_scale_log',
)
Docstring:
Computes the mel spectrogram given the audio filename.
When the parameter `npy_file` is specified, the data is saved to disk as a numpy array (.npy).
Use the parameter `force` to overwrite the numpy array in case it already exists.
The rest of parameters are directly mapped to Essentia algorithms as explained below.
...
```

Reply:
agree that keeping the explicit args for the documentation is a good idea